src/tools/unicode-table-generator/src/main.rs

```diff
@@ -245,56 +243,52 @@ fn main() {
         }
 
         modules.push((property.to_lowercase().to_string(), emitter.file));
-        table_file.push_str(&format!(
-            "// {:16}: {:5} bytes, {:6} codepoints in {:3} ranges (U+{:06X} - U+{:06X}) using {}\n",
+        writeln!(
+            table_file,
+            "// {:16}: {:5} bytes, {:6} codepoints in {:3} ranges (U+{:06X} - U+{:06X}) using {}",
             property,
             emitter.bytes_used,
             datapoints,
             ranges.len(),
             ranges.first().unwrap().start,
             ranges.last().unwrap().end,
             emitter.desc,
-        ));
+        );
         total_bytes += emitter.bytes_used;
     }
     let (conversions, sizes) = case_mapping::generate_case_mapping(&unicode_data);
     for (name, size) in ["to_lower", "to_upper"].iter().zip(sizes) {
-        table_file.push_str(&format!("// {:16}: {:5} bytes\n", name, size));
+        writeln!(table_file, "// {:16}: {:5} bytes", name, size);
         total_bytes += size;
     }
-    table_file.push_str(&format!("// {:16}: {:5} bytes\n", "Total", total_bytes));
-
-    // Include the range search function
-    table_file.push('\n');
-    table_file.push_str(include_str!("range_search.rs"));
-    table_file.push('\n');
+    writeln!(table_file, "// {:16}: {:5} bytes\n", "Total", total_bytes);
 
-    table_file.push_str(&version());
-
-    table_file.push('\n');
+    writeln!(table_file, "use super::rt::*;\n");
+    writeln!(table_file, "{}\n", version());
 
     modules.push((String::from("conversions"), conversions));
 
     for (name, contents) in modules {
-        table_file.push_str("#[rustfmt::skip]\n");
-        table_file.push_str(&format!("pub mod {name} {{\n"));
-        for line in contents.lines() {
-            if !line.trim().is_empty() {
-                table_file.push_str("    ");
-                table_file.push_str(line);
-            }
-            table_file.push('\n');
+        writeln!(table_file, "pub mod {name} {{");
+        for line in contents.trim().lines() {
+            writeln!(table_file, "    {line}");
         }
-        table_file.push_str("}\n\n");
+        writeln!(table_file, "}}\n");
     }
 
-    std::fs::write(&write_location, format!("{}\n", table_file.trim_end())).unwrap();
+    let test_file = generate_tests(&unicode_data);
+
+    std::fs::write(&test_path, test_file).unwrap();
+    std::fs::write(&data_path, table_file).unwrap();
+    rustfmt(&data_path);
+    rustfmt(&test_path);
 }
 
-fn version() -> String {
-    let mut out = String::new();
-    out.push_str("pub const UNICODE_VERSION: (u8, u8, u8) = ");
+fn rustfmt(path: &str) {
+    std::process::Command::new("rustfmt").arg(path).status().expect("rustfmt failed");
```

Comment:
This is surprising to me, this will pick up whichever `rustfmt` happens to be in `PATH`? That we run `rustfmt` or not on the generated code is not the problem, the problem is that we need to run the same `rustfmt` for this generated code and also for the rest of standard library.